From b6b586f3d099dff7c56b69c824a1931ddad170a4 Mon Sep 17 00:00:00 2001
From: Arnaud Ferraris <arnaud.ferraris@collabora.com>
Date: Tue, 23 Apr 2024 16:04:08 +0200
Subject: [PATCH] lib: meson.build: restore libqrtr SONAME

The previous version of `libqrtr` had `libqrtr.so.1` as their SONAME,
while the current version have only `libqrtr.so`. This is problematic as
previously built binaries using this lib will have to be rebuilt for
this new version (or rather, re-linked, which is practically
equivalent).

This change ensures the SONAME is kept by setting the project version
in the top-level `meson.build` and uses this string as the shared
library version.

In practice, the generated library filename is now `libqrtr.so.1.1` with
SONAME still being `libqrtr.so.1`.
---
 lib/meson.build | 1 +
 meson.build     | 1 +
 2 files changed, 2 insertions(+)

diff --git a/lib/meson.build b/lib/meson.build
index 7a894b3..aab46cb 100644
--- a/lib/meson.build
+++ b/lib/meson.build
@@ -5,6 +5,7 @@ pkg = import('pkgconfig')
 libqrtr_srcs = ['logging.c', 'qmi.c', 'qrtr.c']
 libqrtr = shared_library('qrtr',
                          libqrtr_srcs,
+                         version: meson.project_version(),
                          include_directories : inc,
                          install: true)
 
diff --git a/meson.build b/meson.build
index 4478263..0afaf84 100644
--- a/meson.build
+++ b/meson.build
@@ -2,6 +2,7 @@
 
 project('qrtr',
         'c',
+        version: '1.1',
         license : [ 'BSD-3-Clause'],
         default_options : [
         'warning_level=1',
-- 
2.46.2

